packages/blockchain-utils-linking/src/near.ts

```diff
@@ -0,0 +1,51 @@
+import { AccountID } from 'caip';
+import { AuthProvider } from './auth-provider';
+import { getConsentMessage, LinkProof } from './util';
+import { hash } from '@stablelib/sha256';
+import * as uint8arrays from 'uint8arrays';
+
+const stringEncode = (str: string): string => uint8arrays.toString(uint8arrays.fromString(str), 'base64pad');
+
+// REF: near.org
+export class NearAuthProvider implements AuthProvider {
+  readonly isAuthProvider = true;
+
+  constructor(
+    private readonly provider: any,
+    // NOTE: address here is: ed25519 public key
+    private readonly address: string,
+    private readonly chainRef: string) {}
+
+  async authenticate(message: string): Promise<string> {
+    const encodedMsg = stringEncode(message);
+    const { signature } = await this.provider.sign(encodedMsg);
```

Comment:
Did you mention on discord that you had some issue with this signature method not being deterministic?

Reply:
That was another person doing a similar project. They were trying to solve multiple associations of ed25519 key pairs that make up NEAR accounts.

Reply:
I see, thanks for claritying.